### PR TITLE
Fixed instanceX and instanceY variables to be deserialized correctly

### PR DESCRIPTION
**Pull request by DallonTG**
See issue #325. @trharoldsen was right - the instanceX and instanceY variables weren't being deserialized correctly. This PR fixes that.

### CODE REVIEW

**Comment by nelsobe** [comment]
Hesitate to ask :-), but can you tell us the differences between this and what we had before your previous PR that created this problem?  Also, it might be good to reference that PR (as well as your PR that reversed the previous mod) in this one so we have a record of their relationship.

**Comment by DallonTG** [comment]
@nelsobe Are you thinking of the Tincr issue?

For this PR, the site.setName() method was originally changed with PR #289. Before #289, site.setName() parsed the name for X and Y coordinates in addition to setting the name. This caused errors if the sites weren't named normally (where there is only one X and one Y). So, the parsing was moved out of the site.setName() method. We forgot to update the deserialization with this change though, so when a device was deserialized, the X and Y instances weren't being set properly. 

This PR just fixes the deserialization so the X and Y instance variables (the coordinates) are correct.

File: src/main/java/edu/byu/ece/rapidSmith/device/Site.java [hunk 1]
```diff
@@ -824,6 +824,12 @@ private Site readResolve() {
 			site.possibleTypes = possibleTypes;
 			site.externalWires = externalWires;
 			site.bondedType = bondedType;
+			
+			if (instanceX != 0 || instanceY != 0 || !site.parseCoordinatesFromName(name)) { 
+				site.instanceX = (instanceX != 0) ? instanceX : -1; 
+				site.instanceY = (instanceX != 0) ? instanceX : -1;
```
[the remark this answers is not in the document]
Comment (DallonTG) [reply src/main/java/edu/byu/ece/rapidSmith/device/Site.java]: Thanks for catching this.

File: src/main/java/edu/byu/ece/rapidSmith/device/Site.java [hunk 2]
```diff
@@ -824,6 +824,12 @@ private Site readResolve() {
 			site.possibleTypes = possibleTypes;
 			site.externalWires = externalWires;
 			site.bondedType = bondedType;
+			
+			if (instanceX != 0 || instanceY != 0 || !site.parseCoordinatesFromName(name)) {
```
[the remark this answers is not in the document]
Comment (DallonTG) [reply src/main/java/edu/byu/ece/rapidSmith/device/Site.java]: I hadn't considered the case with zero indices. I'm pretty sure you're correct about Integers.

**Comment by DallonTG** [comment]
Addressed @trharoldsen's requested changes.

**Pull request merged**